### PR TITLE
Attempt to get better logging from pure virtual call crashes

### PR DESCRIPTION
**Pull request by jherico**
Right now we see a number of pure virtual call crashes on exit, but the stack traces from Bugsplat are difficult to use and interpret.  This code should cause a stack trace to be printed to the log prior to exit, allowing for easier diagnosis.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 53cffb8cf0f482ab44cd8073d8f7b364e850b8b0 available at: https://gist.github.com/e13717574db6ad93842ed38145656536<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8038.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8038.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ff8d4883b10eafd167d96f6c63b9817bc848a96e available at: https://gist.github.com/e13717574db6ad93842ed38145656536<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8038.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8038.dmg

**Comment by AndrewMeadows** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: ff8d4883b10eafd167d96f6c63b9817bc848a96e available at: https://gist.github.com/e13717574db6ad93842ed38145656536<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8038.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8038.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: eee1ba09062216b0032110f5b1e6583a3ee4510c available at: https://gist.github.com/e13717574db6ad93842ed38145656536<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8038.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8038.dmg

**Comment by jherico** [comment]
## Testing

When running this build, if you enable the developer menu and select `Developer`->`Crash`->`Pure Virtual Function Call`, the end of the log should contain a detailed stack trace of exactly where the crash happened, similar  to this

```
[06/09 12:05:44] [DEBUG] About to make a pure virtual call
[06/09 12:05:44] [WARNING] Pure virtual function call detected
[06/09 12:05:45] [WARNING] "purecall: 7ff88eacdd98 - 0x32X"
[06/09 12:05:45] [WARNING] "crash::pureVirtualCall: 7ff6161bd800 - 0x31X"
[06/09 12:05:45] [WARNING] "QMetaObject::activate: 6e55ada0 - 0x30X"
[06/09 12:05:45] [WARNING] "QAction::activate: 6db468f0 - 0x29X"
[06/09 12:05:45] [WARNING] "QMenu::actionGeometry: 6dca0520 - 0x28X"
[06/09 12:05:45] [WARNING] "QMenu::actionGeometry: 6dca0520 - 0x27X"
[06/09 12:05:45] [WARNING] "QMenu::mouseReleaseEvent: 6dca5600 - 0x26X"
[06/09 12:05:45] [WARNING] "QWidget::event: 6db7ce20 - 0x25X"
[06/09 12:05:45] [WARNING] "QMenu::event: 6dca1c80 - 0x24X"
[06/09 12:05:45] [WARNING] "QApplicationPrivate::notify_helper: 6db55aa0 - 0x23X"
[06/09 12:05:45] [WARNING] "QApplication::notify: 6db53290 - 0x22X"
[06/09 12:05:45] [WARNING] "QCoreApplication::notifyInternal: 6e53c470 - 0x21X"
[06/09 12:05:45] [WARNING] "QApplicationPrivate::sendMouseEvent: 6db56c50 - 0x20X"
[06/09 12:05:45] [WARNING] "QSizePolicy::QSizePolicy: 6dba2520 - 0x19X"
[06/09 12:05:45] [WARNING] "QSizePolicy::QSizePolicy: 6dba2520 - 0x18X"
[06/09 12:05:45] [WARNING] "QApplicationPrivate::notify_helper: 6db55aa0 - 0x17X"
[06/09 12:05:45] [WARNING] "QApplication::notify: 6db53290 - 0x16X"
[06/09 12:05:45] [WARNING] "QCoreApplication::notifyInternal: 6e53c470 - 0x15X"
[06/09 12:05:45] [WARNING] "QGuiApplicationPrivate::processMouseEvent: 7ff884b4e200 - 0x14X"
[06/09 12:05:45] [WARNING] "QGuiApplicationPrivate::processWindowSystemEvent: 7ff884b50800 - 0x13X"
[06/09 12:05:45] [WARNING] "QWindowSystemInterface::sendWindowSystemEvents: 7ff884b36700 - 0x12X"
[06/09 12:05:45] [WARNING] "QEventDispatcherWin32::processEvents: 6e580e60 - 0x11X"
[06/09 12:05:45] [WARNING] "DispatchMessageW: 7ff8b8ac0ae0 - 0x10X"
[06/09 12:05:45] [WARNING] "DispatchMessageW: 7ff8b8ac0ae0 - 0x9X"
[06/09 12:05:45] [WARNING] "QEventDispatcherWin32::processEvents: 6e580e60 - 0x8X"
[06/09 12:05:45] [WARNING] "QEventDispatcherWin32::processEvents: 6e580e60 - 0x7X"
[06/09 12:05:45] [WARNING] "QEventLoop::exec: 6e538fe0 - 0x6X"
[06/09 12:05:45] [WARNING] "QCoreApplication::exec: 6e53b520 - 0x5X"
[06/09 12:05:45] [WARNING] "main: 7ff6161b4300 - 0x4X"
[06/09 12:05:45] [WARNING] "WinMain: 7ff616641e90 - 0x3X"
[06/09 12:05:45] [WARNING] "__tmainCRTStartup: 7ff61661ce10 - 0x2X"
[06/09 12:05:45] [WARNING] "BaseThreadInitThunk: 7ff8baa580e0 - 0x1X"
[06/09 12:05:45] [WARNING] "RtlUserThreadStart: 7ff8bacfc580 - 0x0X"
```

Note the threaded version of this crash doesn't seem to work properly, and just crashes the application without a bugsplat, but this is the same behavior as the current production code.

File: interface/src/CrashReporter.cpp [hunk 1]
```diff
@@ -77,13 +81,43 @@ BOOL redirectLibraryFunctionToFunction(char* library, char* function, void* fn)
     return bRet;
 }
 
+void printStackTrace(ULONG framesToSkip = 1) {
+    QString result;
+    unsigned int   i;
+    void         * stack[100];
```
Comment (ZappoMan) [inline interface/src/CrashReporter.cpp]: coding standard says: `void* stack[100];`

File: interface/src/CrashReporter.cpp [hunk 2]
```diff
@@ -77,13 +81,43 @@ BOOL redirectLibraryFunctionToFunction(char* library, char* function, void* fn)
...
+    unsigned short frames;
+    SYMBOL_INFO  * symbol;
```
Comment (ZappoMan) [inline interface/src/CrashReporter.cpp]: coding standard says no columnar formatting and type * squish

**Comment by hifi-gustavo** [comment]
Build console output for commit: cd7c6be5906019c64a25876b33fd460f7e9eeefa available at: https://gist.github.com/e13717574db6ad93842ed38145656536<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8038.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8038.dmg

**Comment by Nex-Pro** [comment]
big old **PASS**

Log attached.

[hifi-log.txt](https://github.com/highfidelity/hifi/files/309255/hifi-log.txt)

**Pull request merged**